### ISSUE TITLE
selector-max-empty-lines false positive in less mixin

### ISSUE THREAD

**Issue by webOS101** [stylelint/stylelint#2748]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

False positive where `:global()` is used

> Which rule, if any, is this issue related to?

selector-max-empty-lines

> What CSS is needed to reproduce this issue?

```css
.test({
	.hello {
	}

	.fred {
	}
})
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-max-empty-lines": 0
  }
}
```

> Which version of stylelint are you using?

7.13.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

stylelint test.less

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, LESS mixin

> What did you expect to happen?

No lines flagged for warning

> What actually happened (e.g. what warnings or errors you are getting)?

The following error was given (note wrong line #, which I think may have an open issue)

```shell
test.less
 3:3  ✖  Expected no more than 0 empty lines   selector-max-empty-lines
```

**Comment by ntwb** [comment]
Thanks @webOS101 for using the template, makes it easy to understand the issue perfectly 😄 

Did you want to create a pull request for this? For now I've labelled it up as _help wanted_

**Comment by webOS101** [comment]
I may, I haven't taken a look at that module yet to see what's involved.

**Comment by webOS101** [comment]
Taking a quick look, the parser is returning this in rule:
```
  selector: '.test({\n\t.hello {\n\t}\n\n\t.fred {\n\t}\n})',
```
Clearly, that's not right.  Is this an upstream issue or is this something you want to work around in the rule itself?

**Comment by webOS101** [comment]
A quick and dirty solution is to remove parameters from selector string so they won't be considered.  However, this would mean the parameters themselves never got checked.  I suspect that a lot of other rules may be similarly affected.  What may be necessary is to extract all the parameter bodies and have each body validated separately from the surrounding code.

**Comment by jeddy3** [comment]
@webOS101 Thanks for reporting the issue. The built-in rules [ignore non-standard](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#write-the-rule) non-standard constructs, where possible, by the use of the [`isStandardSyntax*`](https://github.com/stylelint/stylelint/tree/master/lib/utils) whitelisting utils.

In this instance, it looks like `selector-max-empty-lines` isn't making use of the [`isStandardSyntaxRule` util](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxRule.js), which would exclude Less mixins. 

Adding it around line [29](https://github.com/stylelint/stylelint/blob/master/lib/rules/selector-max-empty-lines/index.js#L29) should resolve this issue.

> What may be necessary is to extract all the parameter bodies and have each body validated separately from the surrounding code.

This behaviour would exist in a 3rd party plugin, perhaps called `at-mixin-max-empty-lines`. It could possibly live in a `stylelint-less` plugin _pack_. I know there is a [`stylelint-scss`](https://github.com/kristerkari/stylelint-scss) plugin pack, but I don't believe there is a plugin pack for Less-specific features yet. Feel free to create one!

**Comment by webOS101** [comment]
I was thinking more holistically...  The body of all those less mixins should have every (enabled) rule applied to them, not just the selector empty lines rule.  I'll take a look at the plugin pack and see what's available there and how it may work.

**Comment by jeddy3** [comment]
> `selector: '.test({\n\t.hello {\n\t}\n\n\t.fred {\n\t}\n})',` Clearly, that's not right.  Is this an upstream issue or is this something you want to work around in the rule itself?

Good point. Is this the behaviour of `postcss-less@1.1.0`? `stylelint@8.0.0` was released earlier today and it includes `postcss-less@1.1.0`.

**Comment by webOS101** [comment]
I will pull the latest code and retest.

**Comment by webOS101** [comment]
Appears to be the same.  I'll check out `postcss-less` and see if there are any clues there.

**Comment by webOS101** [comment]
I have filed https://github.com/shellscape/postcss-less/issues/86
